src/lib/completion_test.rs

```diff
@@ -0,0 +1,90 @@
+use std::fs;
+use std::path::Path;
+use std::io::Cursor;
+
+#[cfg(test)]
+mod tests {
+    use crate::completion::generate_completion_zsh;
+
+    use super::*;
+    
+    // Function to clean up test environment by removing the completion file
+    fn cleanup() {
+        let home_dir = std::env::var("HOME").expect("Failed to get HOME");
+        let completion_file = format!("{}/.zfunc/_cargo-make", home_dir);
+        if Path::new(&completion_file).exists() {
+            fs::remove_file(&completion_file).expect("Failed to clean up test file");
+        }
+    }
+
+    #[test]
```

Comment:
cargo make splits all tests to 2 parts
multi threaded
single threaded

your tests need to do a cleanup before each invocation so multi threaded won't work
to trigger them only in signle threaded (all tests run at the end...) you need to add ignore flag (see many tests of mine) which will postpone them to the single thread flow

Reply:
Added #[ignore] to make the tests run single threaded